### PR TITLE
Update to .NET 8 preview 1 build

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Update to the latest version of .NET 8 preview 1. Update all the related files.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj [hunk 1]
```diff
@@ -16,25 +16,25 @@
...
   <ItemGroup>
     <!-- This section is to force the version of non-direct dependencies -->
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Extensions.ObjectPool" Version="7.0.3" />
-    <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
-    <PackageReference Include="System.Web.Services.Description" Version="4.10.0" />
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0-preview.1.23110.8" />
+    <PackageReference Include="Microsoft.Extensions.ObjectPool" Version="8.0.0-preview.1.23112.2" />
+    <PackageReference Include="Microsoft.Win32.Registry" Version="6.0.0-preview.5.21301.5" />
+    <PackageReference Include="System.Web.Services.Description" Version="6.0.0-preview1.23060.3" />
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj:22]:
```suggestion
```
These shouldn't be needed

**Review by TravisEz13 — APPROVED** [review]

File: Analyzers.props [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project>
   <ItemGroup>
     <PackageReference Include="DotNetAnalyzers.DocumentationAnalyzers" Version="1.0.0-beta.59" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0" PrivateAssets="all" />
+    <!--PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0" PrivateAssets="all" /-->
```
Comment (adityapatwardhan) [inline Analyzers.props:4]:
```suggestion
```

**Comment by xtqqczze** [comment]
@daxian-dbw  Maybe merge #18877 first?

File: Analyzers.props [hunk 3]
```diff
@@ -1,7 +1,6 @@
 <Project>
   <ItemGroup>
     <PackageReference Include="DotNetAnalyzers.DocumentationAnalyzers" Version="1.0.0-beta.59" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0" PrivateAssets="all" />
```
Comment (daxian-dbw) [inline Analyzers.props:4]: Why is this removed?

File: src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj [hunk 4]
```diff
@@ -16,25 +16,21 @@
 
   <ItemGroup>
     <!-- This section is to force the version of non-direct dependencies -->
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Extensions.ObjectPool" Version="7.0.3" />
-    <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
-    <PackageReference Include="System.Web.Services.Description" Version="4.10.0" />
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj:22]: Why are they removed?

File: Analyzers.props (hunk 3 above)
Comment (adityapatwardhan) [reply Analyzers.props:4]: It is part of the SDK now. Build throws an error since the version in SDK is higher than this.

File: src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj (hunk 4 above)
Comment (adityapatwardhan) [reply src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj:22]: These are transitive dependencies, we do not need to add those anymore. They ship with the SDK and the latest version is auto-included.

File: Analyzers.props (hunk 3 above)
Comment (daxian-dbw) [reply Analyzers.props:4]: That's good to know. Thanks!

File: src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj (hunk 4 above)
Comment (daxian-dbw) [reply src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj:22]: That's good to know, thanks!

**Review by daxian-dbw — APPROVED** [review]
LGTM
